code/game/machinery/kitchen/cooking_machines/_appliance.dm

```diff
@@ -389,14 +389,15 @@
 		CI.result_type = 4//Recipe type, a specific recipe will transform the ingredients into a new food
 		var/list/results = recipe.make_food(C)
 
-		var/atom/temp = new /atom //To prevent infinite loops, all results will be moved into a temporary location so they're not considered as inputs for other recipes
+		var/obj/temp = new /obj(src) //To prevent infinite loops, all results will be moved into a temporary location so they're not considered as inputs for other recipes
```

Comment:
fixes bad loc error. switched to obj at lohikar's request, mostly for standardisation